src/WebJobs.Extensions.DurableTask/EntityScheduler/MessageSorter.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// provides message ordering and deduplication of request messages (operations or lock requests)
+    /// that are sent to entities, from other entities, or from orchestrations.
+    /// </summary>
+    [JsonObject(MemberSerialization.OptIn)]
```

Comment:
I'm a bit confused as to if/when we use JsonSerialization to transfer this across entities. Is this just for durability if an entity has to start back up?

Reply:
Exactly. Entities can go back and forth between memory and storage.